agent/client/channel/channel.go

```diff
@@ -188,10 +188,12 @@ func (c *Channel) send(msg *agentpb.AgentMessage) {
 	}
 
 	// do not use default compact representation for large/complex messages
-	if size := proto.Size(msg); size < 100 {
-		c.l.Debugf("Sending message (%d bytes): %s.", size, msg)
-	} else {
-		c.l.Debugf("Sending message (%d bytes):\n%s\n", size, prototext.Format(msg))
+	if c.l.Logger.IsLevelEnabled(logrus.DebugLevel) {
+		if size := proto.Size(msg); size < 100 {
+			c.l.Debugf("Sending message (%d bytes): %s.", size, msg)
```

Comment:
sorry, you have missed the main point of this PR.
Yes, you right - `Debugf` checks whether DEBUG level is enabled. 
BUT before this check is performed the args for `Debugf` are being evaluated -> the method `prototext.Format(msg)` is called.

So the logic here is:
1. `prototext.Format(msg)` is called
2. `Debugf` checks if debug level is enabled (or in other words "checks whether the already formatted msg from p1 is needed at all". If no - return.
3. Debugf` prints log message.

In case debug level is disabled call p1 wasting resources.

Reply:
Oh, I see, fair enough. 
Since the other devs might be also confused in the future about this check and may decide to remove it as redundant, it might be useful to leave a comment explaining why this check is necessary.